mongo/options/createcollectionoptions.go

```diff
@@ -144,6 +150,12 @@ func (c *CreateCollectionOptions) SetCollation(collation *Collation) *CreateColl
 	return c
 }
 
+// SetChangeStreamPreAndPostImages sets the value for the ChangeStreamPreAndPostImages field.
+func (c *CreateCollectionOptions) SetChangeStreamPreAndPostImages(csppi interface{}) *CreateCollectionOptions {
+	c.ChangeStreamPreAndPostImages = &csppi
```

Comment:
Maybe I missed something, but why do we need `&csppi` here rather than `csppi` directly? I noticed a similar case on [L179](https://github.com/mongodb/mongo-go-driver/blob/b31ec76eccbc5349b7f83079b00192f28b1a4664/mongo/options/createcollectionoptions.go#L179), but a counterexample on [L197](https://github.com/mongodb/mongo-go-driver/blob/b31ec76eccbc5349b7f83079b00192f28b1a4664/mongo/options/createcollectionoptions.go#L197)

Reply:
A great question; I brought it up offline. I _think_ because of automatic dereferencing, the difference between `&csppi` and `csppi` is usually irrelevant in these setters. That's said, it would be good to be consistent across our setters... Filed [GODRIVER-2418](https://jira.mongodb.org/browse/GODRIVER-2418) to pick a consistent style. 

It's _likely_ that passing the pointer is unnecessary. @matthewdale's guess as to why some setters pass the pointer: 

>The driver uses pointer fields to allow "nil means unset" semantics in a lot of places, so I was assuming someone unintentionally followed the same pattern without realizing they were assigning to an interface{} field.

That seems to be true, but passing the pointer as opposed to the value would allow potential modifications to `csppi` later in the code (i.e. `Unmarshal`ing for example). I'll probably keep this as `&csppi` for now, but thanks for pointing out the inconsistency.

Reply:
👍